package.json

```
@@ -1921,10 +1921,10 @@
                     "description": "Set of debug commands to put as code lens above a cell while debugging.",
                     "scope": "resource"
                 },
-                "python.dataScience.ptvsdDistPath": {
+                "python.dataScience.debugpyDistPath": {
```

We actually have a deprecationMessage that we can set on the old setting so that it will not show up in the UI, but users will have a warning if it's already in their json file. I think we would want to use that here.

runMagicCommands has an example of this.

---

@karthiknadig Are we trying to totally hide the ptvsd name in our code base? Or can we leave it in with a deprecation message?

---

When this branch gets merged, there wont be `ptvsd` shipped with the extension. All internal mechanisms that dealt with launching `ptvsd` are removed. The only thing that still works is attached with a caveat, that we show a deprecation message when we detect user attached to `ptvsd`. Looking at how this setting is used, it is adding the path given by user to sys path. So it does not directly do anything with `ptvsd`. But, it will have no effect, since DS will try to  import `debugpy`.

Instead of silently failing i think we need to fail visibly, so people who care about this know that this has changed

---

At this point this branch will probably be merged in the next release. So you could put the deprecation message if you really need it in master. And when this branch merges, it will get rid of it.

---

@IanMatthewHuff  I added the deprecationMessage for the old setting in #12151.

---

Maybe we should not even have this setting? Not sure we need it. Once we create a setting, we can't ever eliminate it. 

If you just remove it from the package.json, we can leave it in the code (makes less changes for you), but then we won't have customers using it.

---
In reply to: [440454458](https://github.com/microsoft/vscode-python/pull/12105#discussion_r440454458) [](ancestors = 440454458)